File: diploma.client/src/models/Product.ts
import ConsensusEvaluation from "./ConsensusEvaluation";
import ExpertEvaluation from "./ExpertEvaluation";

export default class Product {
    asin: string;
    title: string;
    imgUrl: string;
    productUrl: string;
    stars: number;
    reviews: number;
    price: number;
    listPrice: number;
    categoryId: number;
    isBestSeller: boolean;
    boughtLastMonth: number;
    discount: number;
    logReviews: number;
    logPrice: number;
    expertEvaluations?: ExpertEvaluation[] | null;
    consensusEvaluation?: ConsensusEvaluation | null;

    constructor() {
        this.asin = "";
        this.title = "";
        this.imgUrl = "";
        this.productUrl = "";
        this.stars = 0;
        this.reviews = 0;
        this.price = 0;
        this.listPrice = 0;
        this.categoryId = 0;
        this.isBestSeller = false;
        this.boughtLastMonth = 0;
        this.discount = 0;
        this.logReviews = 0;
        this.logPrice = 0;
        this.expertEvaluations = null;
        this.consensusEvaluation = null;
    }
}

File: Diploma.Server/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace Diploma.Server.Models
{
    public class Product
    {
        [Key]
        public string Asin { get; set; }
        public string Title { get; set; }
        public string? ImgUrl { get; set; }
        public string? ProductUrl { get; set; }
        public double? Stars { get; set; }
        public int? Reviews { get; set; }
        public double Price { get; set; }
        public double? ListPrice { get; set; }
        public int? CategoryId { get; set; }
        public bool? IsBestSeller { get; set; }
        public double? BoughtLastMonth { get; set; }
        public double? Discount { get; set; }
        public double? LogReviews { get; set; }
        public double? LogPrice { get; set; }
        public ICollection<ExpertEvaluation>? ExpertEvaluations { get; set; }
        public ConsensusEvaluation? ConsensusEvaluation { get; set; }

        public Product() 
        {
            Asin = "";
            Title = "";
        }
        public Product(Product other)
        {
            Asin = other.Asin;
            Title = other.Title;
            ImgUrl = other.ImgUrl;
            ProductUrl = other.ProductUrl;
            Stars = other.Stars;
            Reviews = other.Reviews;
            Price = other.Price;
            ListPrice = other.ListPrice;
            CategoryId = other.CategoryId;
            IsBestSeller = other.IsBestSeller;
            BoughtLastMonth = other.BoughtLastMonth;
            Discount = other.Discount;
            LogReviews = other.LogReviews;
            LogPrice = other.LogPrice;
        }
    }
}
